src/v2/Components/Onboarding/Steps/Budget.tsx

```diff
@@ -1,18 +1,38 @@
-import React from "react"
+import React, { useState } from "react"
+import { ContextModule, OwnerType } from "@artsy/cohesion"
 import { commitMutation, graphql } from "react-relay"
 import styled from "styled-components"
 import { ProgressIndicator } from "v2/Components/ProgressIndicator"
 import { BudgetUpdateMyUserProfileMutation } from "v2/__generated__/BudgetUpdateMyUserProfileMutation.graphql"
-import { SystemContextProps, withSystemContext } from "v2/Artsy"
+import { withSystemContext } from "v2/Artsy"
 import Colors from "../../../Assets/Colors"
 import { MultiButtonState } from "../../Buttons/MultiStateButton"
 import { media } from "../../Helpers"
-import { Props } from "../Wizard"
 import SelectableToggle from "../SelectableToggle"
-import { StepProps } from "../Types"
 import { Layout } from "./Layout"
-import track from "react-tracking"
-import Events from "../../../Utils/Events"
+import { useTracking } from "v2/Artsy/Analytics/useTracking"
+
+const updateUserProfile = (priceRangeMax, relayEnvironment) => {
+  const input = {
+    priceRangeMin: -1,
+    priceRangeMax,
+  }
+
+  commitMutation<BudgetUpdateMyUserProfileMutation>(relayEnvironment, {
+    mutation: graphql`
+      mutation BudgetUpdateMyUserProfileMutation(
+        $input: UpdateMyProfileInput!
+      ) {
+        updateMyUserProfile(input: $input) {
+          user {
+            name
+          }
+        }
+      }
+    `,
+    variables: { input },
+  })
+}
```

Comment:
Wanted to bring up [this comment](https://github.com/artsy/force/pull/7048#discussion_r579385968) around this -- also would be good to get it into a more conventional form, with error handling, etc, as seen in Sepand's pr (and elsehwere).

Reply:
Ok I think what you're raising here is the lack of `onComplete` and `onError` handing, correct? If so this is another example where I've inherited this code and it's no worse than what was already here. My thought process was that I should improve what I am running into but not really change behavior so I didn't go down the path of "what should we do if the mutations in onboarding fail?" I _think_ what would currently happen is that the mutations would silently fail and the steps would advance through the process anyway. What do you think about kicking this can down the road?

Reply:
Np at all! From a UX perspective that makes sense.